client/extensions/woocommerce/state/sites/shipping-methods/selectors.js

```diff
@@ -3,24 +3,50 @@
 /**
  * External dependencies
  */
-
-import { find, get, isArray } from 'lodash';
+import { translate } from 'i18n-calypso';
+import { filter, get, isArray } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import createSelector from 'lib/create-selector';
 import { getSelectedSiteId } from 'state/ui/selectors';
+import { isWcsEnabled } from 'woocommerce/state/selectors/plugins';
 import { LOADING } from 'woocommerce/state/constants';
 
+/*
+ * By default, those methods are called "XXXX (WooCommerce Services)".
+ * We aren't pushing the "WooCommerce Services" brand anywhere in Calypso, so the method names must be changed.
+ */
+const METHOD_NAMES = {
+	wc_services_usps: translate( 'USPS' ),
+	wc_services_canada_post: translate( 'Canada Post' ),
+};
+
 /**
  * @param {Object} state Whole Redux state tree
  * @param {Number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
  * @return {Array} The list of shipping methods, as retrieved from the server. It can also be the string "LOADING"
  * if the methods are currently being fetched, or a "falsy" value if that haven't been fetched at all.
  */
 export const getShippingMethods = ( state, siteId = getSelectedSiteId( state ) ) => {
-	return get( state, [ 'extensions', 'woocommerce', 'sites', siteId, 'shippingMethods' ] );
+	const allMethods = get( state, [
+		'extensions',
+		'woocommerce',
+		'sites',
+		siteId,
+		'shippingMethods',
+	] );
+	if ( ! isArray( allMethods ) ) {
+		return allMethods;
+	}
+	const availableMethods = isWcsEnabled( state, siteId )
+		? allMethods
+		: filter( allMethods, ( { id } ) => ! id.startsWith( 'wc_services' ) );
+	return availableMethods.map( method => ( {
+		...method,
+		title: METHOD_NAMES[ method.id ] || method.title,
```

Comment:
The only potential problem I see with the renaming is our WooCommerce USPS Shipping extension already uses `USPS` as its method title. If the user installs that plugin, they'll see two USPSes.

Same problem for `Canada Post` with the Canada Post extension we offer.

How about we leave the method title alone if it would result in a duplicate title in the array?

Reply:
If the user installs the `USPS` or `Canada Post` plugins, they won't appear in the Calypso UI. For now, the only shipping methods that can be configured in Calypso are the built-in, and the WooCommerce Services methods, so there's no ambiguity possible.

The reason I changed `USPS (WooCommerce Services)` to `USPS` is that, in Calypso, the user doesn't know what WooCommerce Services is, or even what WooCommerce is. We aren't pushing our brand anywhere else, so putting it here would just be confusing.

Reply:
We do expose WooCommerce Services in Calypso already, e.g.

<img width="455" alt="screen shot 2018-01-08 at 10 29 28 am" src="https://user-images.githubusercontent.com/1595739/34685975-f21c9830-f45e-11e7-92df-2005590eaea6.png">

Let's not rename it - I'll start a P2 discussion